docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,385 @@ title = "Plugin migration guide"
+
+Follow the following [steps](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) to get instructions on how to set up plugin signing.
+
+You will still be able to run and develop your plugin during development by running your Grafana [instance in development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode).
+
+### Time series data can now be structured in a wide or many format.
+
+Starting in Grafana 8 time series data can either be structured in either `wide` or `many` format. Prior to this version time series data was delivered in the following format (which is the `many` format).
+
+| time                 | temperature |
+| -------------------- | ----------- |
+| 2021-05-08T07:31:45Z | 23          |
+| 2021-05-08T09:31:45Z | 25          |
+| 2021-05-08T11:31:45Z | 27          |
+
+| time                 | humidity |
+| -------------------- | -------- |
+| 2021-05-08T07:31:45Z | 60       |
+| 2021-05-08T09:31:45Z | 55       |
+| 2021-05-08T11:31:45Z | 55       |
+
+> Data delivered as an array of data frames (time field needs to be repeated).
+
+That made it possible to detect time series data by inspecting the data frame. If it had two fields (time + value) it most likley was time series data. This is no longer possible since the same data can be delivered in the following `wide` format.
+
+| time                 | temperature | humidity |
+| -------------------- | ----------- | -------- |
+| 2021-05-08T07:31:45Z | 23          | 60       |
+| 2021-05-08T09:31:45Z | 25          | 55       |
+| 2021-05-08T11:31:45Z | 27          | 55       |
+
+> Data delivered as one single data frame (the values share the same time field).
+
+If your plugin should support time series data we recommend you to make sure to support both.
+
+### Update react-hook-form from v6 to v7
+
+We have upgrade the react-hook-form from version 6 to version 7. We recommend follow the [react-hook-form-migration-guide](#https://react-hook-form.com/migrate-v6-to-v7/) to get make your forms compatible with that version.
+
+### Update the plugin.json
+
+The property of defining what Grafana version your plugin support has been renamed.
+
+```json
+// before
+{
+"dependencies": {
+    "grafanaVersion": "7.5.x",
+    "plugins": []
+  }
+}
+
+// after
+{
+  "dependencies": {
+    "grafanaDependency": "8.0.0",
+    "plugins": []
+  }
+}
+```
+
+### Update imports to match emotion 11
+
+Grafana uses emotion to manage the styling of the frontend. The emotion package has now been updated which might affect your frontend plugin if you have any custom styling in it. Luckily you only need to update the import statements to get it working in Grafana 8.
+
+```ts
+// before
+import { cx, css } from 'emotion';
+
+// after
+import { cx, css } from '@emotion/css';
+```
+
+### 8.0 deprecations
+
+#### Grafana theme v1
```

Comment:
@jackw do you think we should cover more things in this section?

Reply:
@mckn the only other thing I can think of that we might consider adding here is the addition of `useStyles2`? 🤔

Reply:
ah, that is true!